edu-services/cm-service/cm-api/api/src/java/org/sakaiproject/coursemanagement/api/CourseManagementService.java

```diff
@@ -444,6 +444,14 @@
 	 */
 	public Map<String, String> findSectionRoles(String userEid);
 
+	/**
+	 * Finds the Sections (and roles) for which a user is a member, but filtered to the specified roles
+	 * @param userEid
+	 * @param roles returned map will only contain mappings to these roles
+	 * @return a Map of Section EIDs to roles for the user
+	 */
+	public Map<String, String> findSectionRolesMatchingRoles(String userEid, List<String> roles);
```

Comment:
A new API method could be avoided here by combining with the one above and simply passing in a null roles if one doesn't want to consider roles at all.